compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5762,6 +5762,22 @@ private void checkObjectFunctionInvocationExpr(BLangInvocation iExpr, BObjectTyp
                 names.fromString(Symbols.getAttachedFuncSymbolName(objectType.tsymbol.name.value, iExpr.name.value));
         BSymbol funcSymbol =
                 symResolver.resolveObjectMethod(iExpr.pos, env, funcName, (BObjectTypeSymbol) objectType.tsymbol);
+
+        int indexOfDot = -1;
+        if (funcSymbol == symTable.notFoundSymbol) {
+            indexOfDot = funcName.value.indexOf('.');
```

Comment:
Since `funcName` is a product of 
`objectType.tsymbol.name.value + Names.DOT.value +  iExpr.name.value` created on line 5762. Can't we use those   them instead of calculating `indexOfDot` and `fieldName` again.

Reply:
Yes, we should use iExpr.name
Thank you, I totally missed that.